### ISSUE TITLE
转换为Qalign

### ISSUE THREAD

**Issue by pxf0429** [zhiyuanyou/DeQA-Score#21]
你好，我将desp_type默认值改为description，是否可以看作是Qalign？

**Comment by pxf0429** [comment]
训练脚本
LOAD="../ModelZoo/mplug-owl2-llama2-7b/"
deepspeed --include localhost:$1 --master_port 6688 src/train/train_mem.py \
    --deepspeed scripts/zero3.json \
    --lora_enable True \
    --model_name_or_path $LOAD \
    --version v1 \
    --dataset_type pair \
    --level_prefix "The quality of the image is" \
    --level_names excellent good fair poor bad \
    --softkl_loss False \
    --weight_rank 0 \
    --weight_softkl 0 \
    --weight_next_token 0 \
    --weight_in_level 0 \
    --weight_desp 1.0 \
    --continuous_rating_loss False \
    --closeset_rating_loss False \
    --use_fix_std True \
    --detach_pred_std True \
    --data_paths ../Data-DeQA-Score/PIPAL/metas/train_pipal.json \
    --data_weights 1 \
    --image_folder ../fineData/ \
    --output_dir ./checkpoints/qalign/PIPAL_deqa_lora \
    --image_aspect_ratio pad \
    --group_by_modality_length True \
    --bf16 True \
    --num_train_epochs 3 \
    --per_device_train_batch_size 16 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 1 \
    --evaluation_strategy "no" \
    --save_strategy "no" \
    --learning_rate 2e-5 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 2048 \
    --gradient_checkpointing True \
    --tune_visual_abstractor True \
    --freeze_vision_model False \
    --dataloader_num_workers 4 \
    --lazy_preprocess True \
    --report_to tensorboard

**Comment by zhiyuanyou** [comment]
你好，想要转化为Q-Align，设置以下两个参数即可：

--dataset_type single
--softkl_loss False

**Comment by pxf0429** [comment]
> 你好，想要转化为Q-Align，设置以下两个参数即可：
> 
> --dataset_type single --softkl_loss False

我也尝试过这样，但是和使用上述pair模式效果差很多，您知道是什么原因吗？上述方式也只有交叉熵损失

**Comment by zhiyuanyou** [comment]
这样做没问题，可能Q-Align并不适合你的任务。

**Comment by pxf0429** [comment]
> Hi! Thanks for your great work! I ran train_lora.sh and found that the description loss is always 0. Is this reasonable? Also, I found no difference when testing after training with --data_weights 1:2 or --data_weights 2:1 Thank you for your reply!
> 
> Here's my fine-tuning script
> 
> #!/bin/bash export PYTHONPATH=./:$PYTHONPATH export CUDA_VISIBLE_DEVICES=2
> 
> LOAD="../ModelZoo/mplug-owl2-llama2-7b"
> 
> deepspeed --master_port 6677 src/train/train_mem.py --deepspeed scripts/zero3.json --lora_enable True --model_name_or_path $LOAD --version v1 --dataset_type pair --level_prefix "The quality of the image is" --level_names excellent good fair poor bad --softkl_loss True --weight_rank 1.0 --weight_softkl 1.0 --weight_next_token 0.05 --continuous_rating_loss True --closeset_rating_loss True --use_fix_std True --detach_pred_std True --data_paths ../Data-DeQA-Score/KONIQ/metas/train_koniq_7k.json ../Data-DeQA-Score/KADID10K/metas/train_kadid_8k.json --data_weights 1 1 --image_folder ../Data-DeQA-Score/ --output_dir ./7.10-train/zhiyuan/single-4-zhiyuan-weight1:2 --image_aspect_ratio pad --group_by_modality_length True --bf16 True --num_train_epochs 3 --per_device_train_batch_size 16 --per_device_eval_batch_size 4 --gradient_accumulation_steps 1 --evaluation_strategy "no" --save_strategy "no" --learning_rate 2e-5 --weight_decay 0. --warmup_ratio 0.03 --lr_scheduler_type "cosine" --logging_steps 1 --tf32 True --model_max_length 2048 --gradient_checkpointing True --tune_visual_abstractor True --freeze_vision_model False --dataloader_num_workers 4 --lazy_preprocess True --report_to tensorboard

It's reasonable for the description_loss to always be 0, because the default task_type is 'score', which does not calculate a description loss.

**Comment by zhiyuanyou** [comment]
> 我针对一些细节崩溃海报和正常海报进行推理，发现正常海报比细节崩溃海报的得分更低且大多集中在3.8～4.5之间，这是什么原因呢？我查看了训练数据大多针对摄影图像。 之后，我合理添加了对应excellent、bad、poor的优质海报和细节崩溃海报，lora微调后测试发现所有海报都集中在2～3之间，并且没有区分性，您可以给我一些建议嘛？

可能是训练数据大部分都集中于2-3之间，需要调整一下训练数据。
